components/org.wso2.carbon.identity.oauth.dcr/src/main/java/org/wso2/carbon/identity/oauth/dcr/service/DCRMService.java

```diff
@@ -234,6 +238,44 @@ public Application updateApplication(ApplicationUpdateRequest updateRequest, Str
                 String backChannelLogoutUri = validateBackchannelLogoutURI(updateRequest.getBackchannelLogoutUri());
                 appDTO.setBackChannelLogoutUrl(backChannelLogoutUri);
             }
+            if (CollectionUtils.isNotEmpty(updateRequest.getAud())) {
+                appDTO.setAudiences(updateRequest.getAud().toArray(new String[0]));
+            }
+            if (updateRequest.getTokenEndpointAuthMethod() != null) {
+                if (validateTokenEndpointAuthMethod(updateRequest.getTokenEndpointAuthMethod())) {
```

Comment:
can't we combine the 2 if conditons and get rid of nested ifs

Reply:
otherwise it is better to add the null check inside validateTokenEndpointAuthMethod function

Reply:
Added the null check to validateTokenEndpointAuthMethod